blockchain_connector/ethereum/ethereum_connector.py

```diff
@@ -0,0 +1,252 @@
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import json
+import web3
+import random
+import asyncio
+import logging
+import argparse
+from urllib.parse import urlparse
+import crypto_utils.crypto_utility as crypto_utility
+from avalon_sdk.worker.worker_details import WorkerType
+import avalon_sdk.worker.worker_details as worker_details
+from avalon_sdk.work_order.work_order_params import WorkOrderParams
+
+from avalon_sdk.ethereum.ethereum_work_order \
+    import EthereumWorkOrderProxyImpl
+from avalon_sdk.ethereum.ethereum_worker_registry \
+    import EthereumWorkerRegistryImpl
+from avalon_sdk.ethereum.ethereum_wrapper \
+    import EthereumWrapper
+from avalon_sdk.direct.jrpc.jrpc_worker_registry \
+    import JRPCWorkerRegistryImpl
+
+
+logging.basicConfig(
+    format="%(asctime)s - %(levelname)s - %(message)s", level=logging.INFO)
+
+LISTENER_SLEEP_DURATION = 5  # seconds
+
+
+class EthereumConnector:
+
+    """
+    This class is the bridge between the Ethereum blockchain and the Avalon
+    core. It listens for events generated by the Ethereum blockchain.
+    It handles event data corresponding to the event (eg: workOrderSubmitted
+    and submits requests to Avalon on behalf of the client. The service also
+    invokes smart contract APIs (eg: workOrderComplete).
+    """
+
+    def __init__(self, config):
+
+        self._config = config
+        self._eth_client = EthereumWrapper(config)
+        tcf_home = os.environ.get("TCF_HOME", "../../")
+
+        worker_reg_contract_file = tcf_home + "/" + \
+            config["ethereum"]["proxy_worker_registry_contract_file"]
+        worker_reg_contract_address = \
+            config["ethereum"]["proxy_worker_registry_contract_address"]
+        self._worker_reg_contract_instance = self._eth_client\
+            .get_contract_instance(
+                worker_reg_contract_file, worker_reg_contract_address)
+
+        work_order_contract_file = tcf_home + "/" + \
+            config["ethereum"]["work_order_contract_file"]
+        work_order_contract_address = \
+            config["ethereum"]["work_order_contract_address"]
+        self._work_order_contract_instance = self._eth_client\
+            .get_contract_instance(
+                work_order_contract_file, work_order_contract_address)
+
+    def _retrieve_registry(self, config):
```

Comment:
Please remove this function. Retrieve registry is not used in block chain connector